examples/with-redux-toolkit/components/counter.js

```diff
@@ -1,29 +1,32 @@
-import React from 'react'
-import { createAction } from '@reduxjs/toolkit'
-import { useSelector, useDispatch } from 'react-redux'
+import React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
 
-const useCounter = () => {
-  const count = useSelector(state => state.count)
-  const dispatch = useDispatch()
-  const increment = () => dispatch(createAction('INCREMENT')())
-  const decrement = () => dispatch(createAction('DECREMENT')())
-  const reset = () => dispatch(createAction('RESET')())
-
-  return { count, increment, decrement, reset }
-}
+import { decrement, increment, reset, selectCount } from '../lib/slices/counterSlice';
 
 const Counter = () => {
-  const { count, increment, decrement, reset } = useCounter()
+  const dispatch = useDispatch();
+  const count = useSelector(selectCount);
+
+  function dispatchIncrement() {
```

Comment:
I personally would use arrow functions here, but that's just a personal preference.